File: UI/travle_core/lib/src/models/role_option_response.dart
import 'package:json_annotation/json_annotation.dart';

part 'role_option_response.g.dart';

/// An elevated role the current user may apply for (mirrors the backend
/// `RoleOptionResponse`). The client resolves the target role id by name from
/// this set rather than hardcoding it.
@JsonSerializable()
class RoleOptionResponse {
  RoleOptionResponse({required this.id, required this.name});

  final int id;
  final String name;

  factory RoleOptionResponse.fromJson(Map<String, dynamic> json) =>
      _$RoleOptionResponseFromJson(json);

  Map<String, dynamic> toJson() => _$RoleOptionResponseToJson(this);
}

File: Backend/Travle.Model/Responses/RoleOptionResponse.cs
namespace Travle.Model.Responses
{
    /// <summary>
    /// An elevated role a user may apply for (Curator or Organizer), as id + name. Backs the
    /// application picker so the client never hardcodes a role id — it resolves the target role by
    /// name from the applicable set returned for the current user.
    /// </summary>
    public class RoleOptionResponse
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
    }
}
